arch/arc/kernel/mcount.S

```diff
@@ -0,0 +1,119 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+/*
+ * Function tracing support for ARC
+ *
+ * Copyright (C) 2023 Synopsys, Inc. (www.synopsys.com)
+ */
+
+
+#include <linux/init.h>
+#include <linux/linkage.h>
+#include <asm-generic/export.h>
+#include <asm/asm-macro-64-bit.h>
```

Comment:
shouldn't this be #include <asm/assembler.h> to cater for both 32-bit and 64-bit.

Reply:
You are every right! In this PR I hadn't considered hs5x, but that is fixed in the newer PR (https://github.com/foss-for-synopsys-dwc-arc-processors/linux/pull/151).
I will close this one since ftrace for ARCv2 is now possible (with GCC >= 14) and the new PR has all the changes in two separate commits